lib/controller.dart

```diff
@@ -38,19 +59,58 @@ void init() async {
 }
 
 void onLoginCompleted() async {
+  view.showLoading();
+
   await loadDataFromFirebase();
+  _filteredInteractions = _allInteractions;
+  _filteredComparisonInteractions = _allInteractions;
+
+  computeUniqFieldValues();
+  _selectedTab = _config.tabs.first.id;
+
   view.setNavlinkSelected(_currentNavLink);
   _navLinks['analyse'].render();
+
+  view.hideLoading();
 }
 
 void onLogoutCompleted() async {
   logger.debug('Delete all local data');
 }
 
 void loadDataFromFirebase() async {
-  view.showLoading();
-  logger.debug('Read all required data');
-  view.hideLoading();
+  try {
+    _config = await fb.fetchConfig();
+  } catch (e) {
+    view.showAlert(_errorMessages['parse-config']);
+    logger.error(e);
+    rethrow;
+  }
+
+  try {
+    _allInteractions =
+        await fb.fetchInteractions(_config.data_paths['interactions']);
+  } catch (e) {
+    view.showAlert(_errorMessages['fetch-interactions']);
+    logger.error(e);
+    rethrow;
+  }
+}
+
+void computeUniqFieldValues() {
+  _uniqueFieldValues = Map<String, Set>();
+  _config.filters.forEach((filter) {
+    _uniqueFieldValues[filter.key] = Set();
+  });
```

Comment:
by the looks of it, it's not computing just the unique field values, but also a filter over them. I'm guessing the filters don't change very much, so it wouldn't make much sense to separate this function into 2 (one for filtering, one for computing unique values), in which case, can we rename it to include filtering? something along the lines of: `computeUniqueFilteredFieldValues`?

Reply:
It's my bad variable naming - it is more of computing all the filter options than actually filtering. So it takes all the possible filter options that the user has mentioned, goes through the entire dataset to compute all the possible unique values for each filter option. 

Updated the variable name to 'filterOptions' in the function, will update in the config if this makes sense.

Reply:
Oh, I see. So the user specifies the "things to filter by", and then this method computes the set of values that each filter can have?

If yes, how about we use "filter category" to refer to the filters specified by the user, and "filter value" for the values that each category can take?

Reply:
Yes that is it, the distinction is much better. Fixed